### DISCUSSION TITLE
Boolean Coordinates

### DISCUSSION THREAD

**Discussion by martinitus** [pydata/xarray#4861] [General]
Hi Folks,
I started using xarray a couple of weeks ago after working with bare multidimensional numpy arrays for years. 

### I love it, you are geniuses! ❤️ 

Today I stumbled over a small pitfall, which I think could be avoided:

I am working with arrays that have axes labeled with categorical values and I ended up using True/False as labels for some binary categories:

```python
test = xarray.DataArray(
     numpy.ones((3,2)), 
     dims=["binary","ternary"],
     coords={"ternary":[3,7,9],"binary":[False,True]}
)
```
now came the big surprise, when I wanted to reduce over selections of the data:

```
test.sel(ternary=[9,3,7]) # does exactly what I expect and gives me the correctly permuted 3x2 array
test.sel(binary=[True,False]) # does not do what I expect
```
Instead of using the coordinate values like with the ternary category, it uses the list as boolean mask and hence I get a 3x1 array at the binary=False coordinate.

I assume that this behavior is reasonable in most cases - And I for sure will stop using bools as binary category labels.
That said in the above case the conceptually identical call results in completely different outcome.

My (radical) proposal would be: forbid binary coordinates in general to avoid such confusion.

Curious about your thoughts! Hth,

Marti

**Comment by mathause** [comment]
Thanks for your report. I opened a new issue with this - I think it's more appropriate there. Unfortunately there seems to be no possibility to directly move a discussion over to the issues.
